cmake/custom/CppCheckTarget.cmake

```diff
@@ -0,0 +1,67 @@
+# Add Analyze with CppCheck target if CppCheck is installed
+find_program(CMAKE_CXX_CPPCHECK cppcheck NAMES cppcheck HINTS $ENV{PROGRAMFILES}/cppcheck)
+  
+if(CMAKE_CXX_CPPCHECK)
+  # Check CppCheck version
+  set(CPP_CHECK_CMD ${CMAKE_CXX_CPPCHECK} --version)
+    execute_process(COMMAND ${CPP_CHECK_CMD}
+      WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}
+      RESULT_VARIABLE CPP_CHECK_RESULT
+      OUTPUT_VARIABLE CPP_CHECK_VERSION
+      ERROR_VARIABLE CPP_CHECK_ERROR)
+    
+  # Check if version could be extracted
```

Comment:
What do you mean with 'version' here?

Reply:
I think I wanted to have a registry of versions of software that were used during the build so we can have better visibility. Not sure this belongs in this PR.